terraform/gitlab/gitlab.tf.json.template.py

```diff
@@ -1211,6 +1211,34 @@ def remove_inconsequential_statements(statements: List[JSON]) -> List[JSON]:
                     mounts:
                     - ["/dev/nvme1n1", "/mnt/gitlab", "ext4", ""]
                     rancher:
+                      services:
+                        clamscan:
+                          image: clamav/clamav:0.104
+                          command: |
+                              /bin/sh -c "clamscan \\
+                              --exclude-dir=^/scan/var/lib/system-docker \\
+                              --exclude-dir=^/scan/proc \\
+                              --exclude-dir=^/scan/sys \\
+                              --exclude-dir=^/scan/dev \\
+                              -rz /scan || \\
+                              echo clamscan failed"
+                          volumes:
+                            - /:/scan:ro
+                            - /mnt/gitlab/clamav:/var/lib/clamav:rw
+                          labels:
+                            io.rancher.os.scope: "system"
```

Comment:
What are the choices here and why did you chose this `system`?

Reply:
`user` and `system`, was having issues with cron scheduling when using the `user` configuration, services would not start up at their scheduled time.

Reply:
With the scope set to `"user"`, ros creates the the containers, but does not run them at the scheduled time, there might be some modifications to make to the `container-cron` service if we want the scope set to `"user"`.
```
#cloud-config
rancher:
  services:
    clamscan:
      image: clamav/clamav:0.104
      command: |
          /bin/sh -c "clamscan \\
          --exclude-dir=^/scan/var/lib/system-docker \\
          --exclude-dir=^/scan/proc \\
          --exclude-dir=^/scan/sys \\
          --exclude-dir=^/scan/dev \\
          -rz /scan || \\
          echo clamscan failed"
      volumes:
        - /:/scan:ro
        - /mnt/gitlab/clamav:/var/lib/clamav:rw
      labels:
        io.rancher.os.scope: "user"
        io.rancher.os.createonly: "true"
        cron.schedule: "0 46 0 * * ?"
    freshclam:
      image: clamav/clamav:0.104
      command: /bin/sh -c "freshclam || echo freshclam failed"
      volumes:
        - /:/scan:ro
        - /mnt/gitlab/clamav:/var/lib/clamav:rw
      labels:
        io.rancher.os.scope: "user"
        io.rancher.os.createonly: "true"
        cron.schedule: "0 44 0 * * ?"
```

```
[rancher@ip-172-31-2-83 ~]$ date
Tue Apr 26 00:44:56 UTC 2022
[rancher@ip-172-31-2-83 ~]$ docker ps -a
CONTAINER ID        IMAGE                 COMMAND                  CREATED             STATUS              PORTS               NAMES
75b92900bac4        clamav/clamav:0.104   "/init /bin/sh -c 'f…"   2 minutes ago       Created                                 freshclam
16a4b4bb330f        clamav/clamav:0.104   "/init /bin/sh -c 'c…"   2 minutes ago       Created                                 clamscan
[rancher@ip-172-31-2-83 ~]$ date
Tue Apr 26 00:46:04 UTC 2022
[rancher@ip-172-31-2-83 ~]$ docker ps -a
CONTAINER ID        IMAGE                 COMMAND                  CREATED             STATUS              PORTS               NAMES
75b92900bac4        clamav/clamav:0.104   "/init /bin/sh -c 'f…"   3 minutes ago       Created                                 freshclam
16a4b4bb330f        clamav/clamav:0.104   "/init /bin/sh -c 'c…"   3 minutes ago       Created                                 clamscan
[rancher@ip-172-31-2-83 ~]$
```

Reply:
Should mention that in a comment.

Reply:
You should briefly explain that in a comment in the source code.

Reply:
Update, was able to get a user service running with container-cron. Need to talk to Hannes and see if we should reduce the scope of this job from `system` to `user`. https://github.com/DataBiosphere/azul/pull/4155#issuecomment-1112782295